### DISCUSSION TITLE
Modifier rename

### DISCUSSION THREAD

**Discussion by chrisbanes** [chrisbanes/haze#451] [Polls]
Hi everyone. In v0.9 Haze was completely re-written in a way where the way that Haze works underneath was completely inverted.

Pre-v0.9, all of the drawing and logic was performed by your `Modifier.haze` node, and your `Modifier.hazeChild` nodes just updated some bounds state to tell the `haze` node where to blur. In this scenario, the naming of `haze` + `hazeChild` made sense.

In v0.9+ though, all of the blur effect drawing is now down by the `hazeChild` node. Your `haze` nodes are responsible for drawing their content into a [GraphicsLayer](https://developer.android.com/reference/kotlin/androidx/compose/ui/graphics/layer/GraphicsLayer) for the `hazeChild` nodes to draw. In this instance, the naming is starting to get a bit tenuous.

Fast forward to the upcoming release, and the new ability for nested blurring effects (in #441), and the naming is now just plain wrong. 🫤 

So, should we fix that? Naming is the hardest problem in software, and I thought I'd crowdsource it. I have added a few options that I've been thinking about, but please comment if you can think of something better.

**Comment by almozavr** [comment]
hazeSource + hazeEffect

**Reply by ajsmithsw** [reply]
@almozavr I like hazeSource! But perhaps hazeSource + hazeBackground? When I'm applying hazeChild to a composable it feels like I'm handling the background behaviour, just like if I was setting the .background() modifier, so would feel intuitive to me.

@chrisbanes this library is awesome btw, thank you 🙏

**Reply by almozavr** [reply]
hazeBackground sounds really good to me as well, agree

**Reply by chrisbanes** [reply]
I like `hazeSource` + `hazeBackground`. 

The word background is a little ambiguous (see my comment below), but I think it's fine for this.

**Comment by sergejsha** [comment]
What do you think about naming child items to be fogged up/hazed as `hazed` (or `hazeable`) and the layer where they are shown hazy `hazeBackground`?

The naming could be then 
a) `hazed` + `hazeBackground` or 
b) `hazeable` + `hazeBackground`

**Comment by hrach** [comment]
From my POV anything with "background" makes me puzzled. Is that the "haze" or the "background source" for haze?

**Reply by hrach** [reply]
Therefore hazeSource + hazeEffect seems the best for me.

**Reply by chrisbanes** [reply]
Yeah, it's tricky as both could be called `background` really:

- `haze` captures background content
- `hazeChild` is basically a (very) complicated `Modifier.background`

**Comment by svenjacobs** [comment]
Thinking about it, I would prefer modifier names that clearly state **where** the haze effect is rendered (currently `haze`) and **what** is hazed (`hazeChild`). So I guess ~~`hazeBackdrop`~~ `hazeSource` and `hazeEffect` is the most logical to me.

**Reply by chrisbanes** [reply]
> the haze effect is rendered (currently haze) and what is hazed (hazeChild)

It's actually the opposite 😬. `haze` captures the background content, `hazeChild` draws the effect as a background.

This is why I'm leaning towards `hazeSource` (`haze`) + `hazeEffect` (`hazeChild`), as it quite neatly captures their goals.

**Reply by svenjacobs** [reply]
Dammit, mixed those up. But still the **where** and **what** is important for me 😃

**Reply by chrisbanes** [reply]
Yeah, choosing between a verb vs subject is tricky when it comes to modifiers.

**Reply by sergejsha** [reply]
`hazeSource` + `haze`, 
`hazeUnderlay` + `hazeOverlay`?

Defining a clear hierarchy in the names relieves mental effort needed for understanding where each modifier should be placed.

**Reply by chrisbanes** [reply]
Using `haze` isn't possible right now as it already exists. We can't deprecate something and then rename something else to it.

**Comment by chrisbanes** [comment]
Thanks everyone for the votes and suggestions! Y'all voted for `hazeSource` + `hazeEffect`, which I totally agree with. Implemented in #452
